osu.Game/Screens/Play/BreakPeriodsTrackOverlay.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Allocation;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Timing;
+using osu.Game.Beatmaps.Timing;
+using osu.Game.Graphics;
+using System;
+using System.Collections.Generic;
+
+namespace osu.Game.Screens.Play
+{
+    public class BreakPeriodsTrackOverlay : Container
+    {
+        private const double fade_duration = BreakPeriod.MIN_BREAK_DURATION_FOR_EFFECT / 2;
+        private const int arrows_appear_offset = 1000;
+
+        private readonly BindableDouble healthBindable = new BindableDouble
+        {
+            MinValue = 0,
+            MaxValue = 1
+        };
+
+        private int currentBreakIndex;
+        private double iconAppearTime;
+        private bool isBreak;
+        private bool iconHasBeenShown;
+        private bool arrowsHasBeenShown;
+        private bool startArrowsHasBeenShown;
+        private double health;
+        private readonly double startTime;
+
+        private readonly List<BreakPeriod> breaks;
+
+        private SampleChannel samplePass;
+        private SampleChannel sampleFail;
+
+        private readonly SpriteIcon resultIcon;
+        private readonly ArrowsOverlay arrows;
+
+        private IClock audioClock;
+        public IClock AudioClock { set { audioClock = value; } }
+
+        public Action OnBreakIn;
+        public Action OnBreakOut;
+
+        public BreakPeriodsTrackOverlay(double startTime, List<BreakPeriod> breaks)
+        {
+            this.startTime = startTime;
+            this.breaks = breaks;
+
+            RelativeSizeAxes = Axes.Both;
+            Children = new Drawable[]
+            {
+                resultIcon = new SpriteIcon
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Size = new Vector2(100),
+                    Alpha = 0,
+                },
+                arrows = new ArrowsOverlay(),
+            };
+
+            healthBindable.ValueChanged += newValue => { health = newValue; };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(AudioManager audio)
+        {
+            samplePass = audio.Sample.Get(@"Gameplay/sectionpass");
+            sampleFail = audio.Sample.Get(@"Gameplay/sectionfail");
+        }
+
+        public void BindHealth(BindableDouble health) => healthBindable.BindTo(health);
+
+        protected override void Update()
+        {
+            if (breaks == null) return;

```

Comment:
you can't just overload update to do stuff like this. it's shocking for readability and extensibility.